portal/models/qb_timeline.py

```diff
@@ -786,6 +761,15 @@ def invalidate_users_QBT(user_id, research_study_id):
         for ad in adh_data:
             db.session.delete(ad)
 
+        if not current_app.config.get("TESTING", False):
+            # clear the timeout lock as well, since we need a refresh
+            # after deletion of the adherence data
```

Comment:
```suggestion
            # after deletion of the adherence data
            # otherwise, we experience a deadlock situation where tables can't be dropped 
            # between test runs, as postgres believes a deadlock condition exists
```